### ISSUE TITLE
Search GET endpoint to allow for possibility of bookmarklet

### ISSUE THREAD

**Issue by ThaMunsta** [glitch-soc/mastodon#1909]
### Pitch

There does exist a share endpoint that can be used with a bookmarklet to post about the page you are looking at quickly from your own instance. (This https://rknightuk.github.io/mastodon-toot-bookmarklet/ for example) 

If there was some kind of search endpoint ie: `[url]/search?q=account@someOtherInstance.social` it might be possible to streamline replies/follow/etc when browsing "external" instances. 

I picture going to an instance , finding a post and being able to click the bookmarklet to open a search from _my_ hosted instance and trigger the data to pull in for local interaction. Instead of this "Simply copy and paste this URL into the search bar of your favorite app or the web interface where you are signed in." step...

tl;dr could this help skip the copy/paste/search steps needed today?

### Motivation

There are a lot of new users joining, especially in the last week. If users had an easy way to find their way back to their "home instance" and reply in a pop up or something I think it would streamline the experience. New users don't seem to know what instance they are on or how to follow accounts from other instances. This may not fix things for them but it could be a tool.
